Floofbot/Modules/ModMail.cs

```diff
@@ -0,0 +1,195 @@
+        }
+
+        [Command("channel")]
+        [Summary("Sets the channel for the modmail notifications")]
+        public async Task Channel([Summary("Channel (eg #alerts)")]Discord.IChannel channel)
+        {
+            CheckServerEntryExists(Context.Guild.Id);
+            var ServerConfig = _floofDB.ModMails.Find(Context.Guild.Id);
+            ServerConfig.ChannelId = channel.Id;
+            _floofDB.SaveChanges();
+            await Context.Channel.SendMessageAsync("Channel updated! I will send modmails to <#" + channel.Id + ">");
+        }
+
+        [Command("toggle")]
+        [Summary("Toggles the modmail module")]
+        public async Task Toggle()
+        {
+
+            // try toggling
+            try
+            {
+                CheckServerEntryExists(Context.Guild.Id);
+                // check the status of logger
+                var ServerConfig = _floofDB.ModMails.Find(Context.Guild.Id);
+                if (ServerConfig.ChannelId == null)
+                {
+                    await Context.Channel.SendMessageAsync("Channel not set! Please set the channel before toggling the ModMail feature.");
+                    return;
+                }
+                ServerConfig.IsEnabled = !ServerConfig.IsEnabled;
+                _floofDB.SaveChanges();
+                await Context.Channel.SendMessageAsync("Modmail " + (ServerConfig.IsEnabled ? "Enabled!" : "Disabled!"));
+            }
+            catch (Exception ex)
+            {
+                await Context.Channel.SendMessageAsync("An error occured: " + ex.Message);
+                Log.Error("Error when trying to toggle the modmail: " + ex);
+                return;
+            }
+        }
+        [Command("modrole")]
+        [Summary("OPTIONAL: A Role to Ping When ModMail is Received.")]
+        public async Task SetModRole(string roleName = null)
+        {
+            var ServerConfig = _floofDB.ModMails.Find(Context.Guild.Id);
+            if (roleName == null)
+            {
+                await Context.Channel.SendMessageAsync("", false, new EmbedBuilder { Description = $"💾 Usage: `modmailconfig [rolename]`", Color = GenerateColor() }.Build());
+                return;
+            }
+            foreach (SocketRole r in Context.Guild.Roles)
+            {
+                if (r.Name.ToLower() == roleName.ToLower())
+                {
+                    try
+                    {
+                        ServerConfig.ModRoleId = r.Id;
+                        await Context.Channel.SendMessageAsync("Mod role set!");
+                        _floofDB.SaveChanges();
+                        return;
+                    }
+                    catch (Exception ex)
+                    {
+                        await Context.Channel.SendMessageAsync("An error occured: " + ex.Message);
+                        Log.Error("Error when trying to set the modmail mod role: " + ex);
+                        return;
+                    }
```

Comment:
May want to chuck a `break` in here. Discord allows the existence of multiple roles with the same name. An optional refactor would be to filter the role list first, check to see if there are any duplicates, and raise an error based on that -- but that's pretty overkill for this project.

Reply:
I've added a bool check that is triggered when it finds a role, if it has found a role, it continues checking the role list for other roles with the same name and throws an error if there are